pkg/router/crl/crl.go

```diff
@@ -0,0 +1,238 @@
+package crl
+
+import (
+	"bytes"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	logf "github.com/openshift/router/log"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+var log = logf.Logger.WithName("crl")
+
+const crlFilePermissions = 0600
+
+// authorityKeyIdentifier is a certificate's authority key identifier.
+type authorityKeyIdentifier struct {
+	KeyIdentifier []byte `asn1:"optional,tag:0"`
+}
+
+var authorityKeyIdentifierOID = asn1.ObjectIdentifier{2, 5, 29, 35}
+
+func ManageCRLs(caBundleFilename, crlFilename string, caUpdateChannel <-chan struct{}) error {
+	nextUpdate, err := updateCRLFile(caBundleFilename, crlFilename)
+	if err != nil {
+		return err
+	}
+	go func() {
+		for true {
+			if nextUpdate.IsZero() {
+				// TODO: Make this a longer interval (for example, hourly) or implement backoff logic?
+				log.V(1).Info("Couldn't determine next CRL update time. Falling back to updating every 5 minutes.")
+				nextUpdate = time.Now().Add(5 * time.Minute)
```

Comment:
I currently have it set to 5 minutes to aid in testing, since I manually update the CRL files, and sometimes I've missed the actual expiration time. I will add a constant with the appropriate fallback interval before this is ready to merge.

Reply:
If the failure was transient should we wait for another day before downloading again? I could see a case where we do a number of retries before falling back to downloading once a day. Otherwise (and assuming a transient download failure) the only way to get a new list (in a hurry) is to delete the pod.